### PR TITLE
Tests: Use LocalRest for DNS queries, not bare Channels

### PR DESCRIPTION
**Pull request by Geod24**
```
LocalRest was intended as a way to simplify RPC calls, no mattter the type
of communication going on under the hood, as it simply replaces it.
So instead of using bare Channel for communication, just use the same RPC
approach, by exposing one method for each communication method.
```

Haven't tested it yet, but here's the idea.

### CODE REVIEW

File: source/agora/test/Base.d [hunk 1]
```diff
@@ -931,11 +928,9 @@ public class TestAPIManager
         assert(conf.interfaces.length == 1);
 
         auto time = new shared(TimePoint)(this.initial_time);
-        this.dns_chan = new Channel!DNSQuery();
-        this.dns_resolver = new LocalRestDNSResolver(this.dns_chan);
```
Comment (mkykadir) [inline source/agora/test/Base.d:935]: `test/DNS.d` uses this during tests

Comment (Geod24) [reply source/agora/test/Base.d:935]: Fixed

**Review by mkykadir — APPROVED** [review]
This is cool :sunglasses:

**Comment by omerfirmak** [comment]
We now either need NS records in unittests or move `registry_address` out of `ValidatorConfig`

**Comment by Geod24** [comment]
> We now either need NS records in unittests or move registry_address out of ValidatorConfig

I started doing this, then hit a configy bug, which turned into a DMD bug. So WIP.

**Pull request merged**